crud/interface.go

```diff
@@ -10,6 +10,9 @@ import (
 
 // Gets the current BareMetal Resource
 type ResourceFetcher interface {
+	// Get should update the s.Resource, and is used by crud.ReadResource() to populate s.D
+	// Get() may expect s.D.Id() to be set, but not s.Resource, or anything else in s.D
+	// TODO: switch to expect s.ID() instead of s.D.Id()
```

Comment:
This is totally incorrect, and was based on my misunderstanding the point of s.ID().